server.js

```diff
@@ -24,9 +24,114 @@ app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  // res.send('Hello planet Earth!')
+  res.send("Try searches based on the routes: /books, /books/sort/rating, /booklists?page=2, /books/id/[a number], /author/[name of author], /titles/[name of title] or  query for any word in title or author: library?search=[type your search]")
+
+})
+
+// Example of JSON-structure 
+// {
+//   "bookID": 3,
+//   "title": "Harry Potter and the Sorcerer's Stone (Harry Potter  #1)",
+//   "authors": "J.K. Rowling-Mary GrandPré",
+//   "average_rating": 4.47,
+//   "isbn": 439554934,
+//   "isbn13": 9780439554930,
+//   "language_code": "eng",
+//   "num_pages": 320,
+//   "ratings_count": 5629932,
+//   "text_reviews_count": 70390
+// }
+
+//Parameters-explanained
+//Path variable or path parameter  = something specific; an individual/item or group
+// Easiest way of creating a rest route called books (path variable) (localhost:8080/books)
+app.get('/books', (req, res) => {
+  res.json(booksData)
+})
+
+
+// PAGE - to have the result on a page with x number of boooks.
+
+const PER_PAGE = 10
+
+
+// routte to GET books by page (http://localhost:8080/booklists?page=2)
+app.get('/booklists', (req, res) => {
+  const { page } = req.query
+  const startIndex = PER_PAGE * +page
+  //console.log(startIndex)
+  const data = booksData.slice(startIndex, startIndex + PER_PAGE)
+
+  res.json({
+    totalPages: Math.floor(booksData.length / PER_PAGE),
+    currentPage: +page,
+    data
+  })
 })
 
+//creating a sorted rating (localhost:8080/books/sort/rating)
+app.get('/books/sort/rating', (req, res) => {
```

Comment:
It would be nice if this was part of the `/books` endpoint (and it'd be more RESTful) - then make use of query params to filter on rating. So you could have `/books` to return all books and then `/books?rating=5` etc